src/main/java/bosco/Bosco.java

```diff
@@ -0,0 +1,295 @@
+    private static final String INDENT_EXTRA = "  ";
+    private static final String MESSAGE_MARK_DONE = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_MARK_UNDONE = "OK, I've marked this task as not done yet:";
+    private static final String MESSAGE_ADDED_TASK = "Got it. I've added this task:";
+    private static final String MESSAGE_DELETED_TASK = "Noted. I've removed this task:";
+    private static final String DEADLINE_PREFIX_BY = "/by";
+    private static final String EVENT_PREFIX_FROM = "/from";
+    private static final String EVENT_PREFIX_TO = "/to";
+
+    private static final Scanner CLI_SCANNER = new Scanner(System.in);
+
+    private static final ArrayList<Task> tasksList = new ArrayList<>();
+
+    private static void loadFileContents(String filePath) throws IOException {
+        Path inputPath = createFilePathIfNotExists(filePath);
+        Scanner fileScanner = new Scanner(inputPath);
+        while (fileScanner.hasNext()) {
+            addTaskFromFileLine(fileScanner.nextLine());
+        }
+    }
+
+    private static Path createFilePathIfNotExists(String filePath) {
+        Path p = Paths.get(filePath);
+        try {
+            Files.createDirectories(p.getParent());
+            if (!Files.exists(p)) {
+                Files.createFile(p);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return p;
+    }
+
+    private static void addTaskFromFileLine(String fileLine) {
+        String[] stringParts = fileLine.split(" \\| ");
+        boolean isDone = (stringParts[1].equals("X"));
+        String taskType = stringParts[0];
+        String description = stringParts[2];
+        switch(taskType) {
+        case "T":
+            tasksList.add(new Todo(description, isDone));
+            break;
+        case "D":
+            tasksList.add(new Deadline(description, isDone, stringParts[3]));
+            break;
+        case "E":
+            tasksList.add(new Event(description, isDone, stringParts[3], stringParts[4]));
+            break;
+        }
+    }
+
+    private static void writeToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task task: tasksList) {
+            fw.write(getFileInputForTask(task) + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    private static String getFileInputForTask(Task task) {
+        if (task instanceof Todo) {
+            return "T | " + task.getStatusIcon() + " | " + task.getDescription();
+        } else if (task instanceof Deadline) {
+            return "D | " + task.getStatusIcon() + " | " + task.getDescription()
+                    + " | " + ((Deadline)task).getBy();
+        } else {
+            return "E | " + task.getStatusIcon() + " | " + task.getDescription()
+                    + " | " + ((Event)task).getFrom() + " | " + ((Event)task).getTo();
+        }
```

Comment:
Consider this string formatting to be done within each class, as to maintain all related attributes together and would be easier to manage and scale